homeassistant/components/light/mqtt_json.py

```diff
@@ -180,11 +202,13 @@ def turn_on(self, **kwargs):
         message = {'state': 'ON'}
 
         if ATTR_RGB_COLOR in kwargs:
-            message['color'] = {
-                'r': kwargs[ATTR_RGB_COLOR][0],
-                'g': kwargs[ATTR_RGB_COLOR][1],
-                'b': kwargs[ATTR_RGB_COLOR][2]
-            }
+            message.update(json.loads(self._templates[CONF_RGB_SET_TEMPLATE](
```

Comment:
No, this is too confusing and should be removed.

It also contains a bug: it no longer assigns to `message['color']` but instead updates `message` with arbitrary data.